contributing/documentation-site.md

```diff
@@ -0,0 +1,9 @@
+# Documentation site
+
+We have a new documentation site under `design-system-docs`.
+
+This docs site is deployed via Netlify which allows us to access previews of work in progress changes. You can request a preview deploy for any pull request by adding the "docs preview" label.
```

Comment:
It's worth adding that external collaborators / forked repos will not be able to deploy a preview to Netlify.